insalan/tournament/serializers.py

```diff
@@ -63,36 +63,52 @@ class Meta:
         )
         fields = "__all__"
 
+    def to_representation(self, instance):
+        """Remove all fields except id and is_announced when is_announced is False"""
+        ret = super().to_representation(instance)
+        if ret["is_announced"]:
+            return ret
+        else:
+            return {"id": ret["id"], "is_announced": False}
+
 
 class TeamSerializer(serializers.ModelSerializer):
     """Serializer class for Teams"""
 
-    players = serializers.ListField(required=False, source="get_players_id")
-    managers = serializers.ListField(required=False, source="get_managers_id")
+    players = serializers.ListField(required=False, source="get_players_id", validators=[unique_registration])
+    managers = serializers.ListField(required=False, source="get_managers_id", validators=[unique_registration])
+    password = serializers.CharField(write_only=True)
+    players_pseudos = serializers.ListField(required=False, write_only=True)
+    managers_pseudos = serializers.ListField(required=False, write_only=True)
```

Comment:
It's bad practice to match elements with two lists. Please prefer doing a list of objects, each object contaning the player id and their name

Reply:
I agree, but in this case we only need the pseudos when we serialize a team whereas we need the players id when we serialize and deserialize. If you have a better solution, I am all ears.